src/credentials/CHIPOperationalCredentials.cpp

```diff
@@ -266,6 +266,82 @@ CHIP_ERROR OperationalCredentialSet::SetDevOpCredKeypair(const CertificateKeyId
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR OperationalCredentialSet::ToSerializable(const CertificateKeyId & trustedRootId,
```

Comment:
Is this functionality intended to run on a node? If so, I'd suggest to do more optimization toward a zero-copy implementation

Reply:
By zero-copy, you mean having data pointers to point to the actual data? The same data being pointed to by the OperationalCredentialSet class?
That's a nice suggestion, but I'm afraid that would not be possible because here the data needs to be serialized in memory, so then it can be written over to the PersistentStorage Class